### PR TITLE
nixos/apache-kafka: add mirrormaker options

### PR DESCRIPTION
**Pull request by itihas**
###### Motivation for this change

Kafka clusters can be mirrored (wholly or in part) using a tool provided in the package. Options to run it as a systemd service, and configure which topics are mirrored, have been created.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by illustris — CHANGES_REQUESTED** [review]
Couldn't check if the mirrormaker service can actually mirror data yet, but these changes are needed for it to build successfully.

File: nixos/modules/services/misc/apache-kafka/mirrormaker.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+let
+  cfg = config.services.apache-kafka.mirrormaker;
+  mirrormakerOpts = { name, ... }:
+    {
+      options = with types; {
+        package = mkOption { type = package; default = pkgs.apacheKafka; description = "kafka package this instance refers to"; };
+        topics = mkOption { type = listOf str; description = "topics this mirrormaker instance will mirror"; };
+        heapOpts = mkOption { type = str; description = "heap options to pass to java"; };
+        streams = mkOption { type = int; description = "number of consumer threads"; default = 1; };
```
[the remark this answers is not in the document]
Comment (itihas) [reply nixos/modules/services/misc/apache-kafka/mirrormaker.nix]: did so

**Comment by itihas** [comment]
> do you mind including a trivial test? even something as simple as launching an instance, and showing a port was bound is quite beneficial from a review perspective

done; I added to the existing nixos/tests/kafka.nix. The service now mirrors a topic to a second kafka cluster, where the message must show up.

**Comment by itihas** [comment]
UM. sorry.

**Pull request closed without merging**